app/models/agents/delay_agent.rb

```diff
@@ -55,11 +57,17 @@ def receive(incoming_events)
 
     def check
       if memory['event_ids'] && memory['event_ids'].length > 0
-        received_events.where(id: memory['event_ids']).reorder('events.id asc').each do |event|
+        events = received_events.where(id: memory['event_ids']).reorder('events.id asc')
+
+        if options['max_emitted_events'].present?
+          events = events.slice(0, options['max_emitted_events'].to_i)
```

Comment:
I think it is more efficient to use limit() here: `events = events.limit(options['max_emitted_events'].to_i)`